pkg/config/config.go

```diff
@@ -321,6 +322,14 @@ func defaultConfig() *Config {
 	}
 }
 
+// NewConfigFromConfigMap returns a Config for the given configmap
+func NewConfigFromConfigMap(config *corev1.ConfigMap) (*Config, error) {
+	if config == nil {
+		return NewConfigFromMap(nil)
```

Comment:
```suggestion
                return defaultConfig(), nil
```

It depends on the person but I think it is better to simply return `defaultConfig()`.

Reply:
Another code like `leaderelection` has it https://github.com/knative/pkg/blob/main/leaderelection/config.go#L74-L76

Reply:
@nak3 

hmm interesting, since the beginning of NewConfigFromMap uses defaultConfig anyway, why not just skip all the extra stuff and return it directly?

I think I see your point. However, for now I will leave it as is, my reason being:

There are quite a few different checks in the NewConfigFromMap, and I feel safer having then run no matter if there is a configmap or not. Then, on the chance we change something about the default config, we have more code checks in place to yell at us if we end up changing a behavior we weren't expecting.

However, I'm happy to hear your response, I can always open a new PR later if you don't think my concern would actually be a problem 👍